verilog/preprocessor/verilog_preprocess.cc

```diff
@@ -613,13 +613,12 @@ absl::Status VerilogPreprocess::HandleInclude(
   verilog::VerilogPreprocess child_preprocessor(config_);
   // TODO(karimtera): share the macros defined with child pp.
   // TODO(karimtera): limit number of nested pps.
-  absl::string_view source_contents_view = source_contents;
-  preprocess_data_.child_included_content.emplace_back(std::make_unique<verible::TextStructure>(source_contents_view));
+  preprocess_data_.child_included_content.emplace_back(std::make_unique<verible::TextStructure>(source_contents));
   verible::TextStructure& included_structure = *preprocess_data_.child_included_content.back();
   verible::TokenSequence& included_sequence = included_structure.MutableData().MutableTokenStream();
   verible::TokenStreamView& included_view = included_structure.MutableData().MutableTokenStreamView();
 
-  verilog::VerilogLexer lexer(source_contents);
+  verilog::VerilogLexer lexer(included_structure.Data().Contents());
```

Comment:
This was the problem, the `TextStructure` mainly contains 3 things:

1. `string owned_contents_` which owns the memory that should be referenced by tokens.
2. `TokenSequence tokens_` the lexed tokens.
3. `TokenStreamView tokens_view_` a modified view of `tokens_`, in this case the preprocessed view.

The problem was that `tokens_` pointing to `source_contents` which goes out of scope after `HandleInclude` finishes.

Reply:
Update: 
`TokenStreamView tokens_view_` is left empty, because there is a restriction that it should only refer to tokens of the same text structure.